packages/workspace/src/parser/internal/native/helpers.ts

```diff
@@ -0,0 +1,113 @@
+/*
+*                      Copyright 2020 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+import { ReferenceExpression, ElemID, Value, TypeElement, ListType, ObjectType, PrimitiveTypes } from '@salto-io/adapter-api'
+import isPromise from 'is-promise'
+import { LexerToken } from './lexer'
+import { SourcePos, IllegalReference, SourceRange } from '../types'
+import { ParseContext } from './types'
+import { Keywords } from '../../language'
+
+export const getPosition = (token: LexerToken, startOfToken = true): SourcePos => {
+  const startPos = {
+    col: token.col,
+    line: token.line,
+    byte: token.offset,
+  }
```

Comment:
naming nit: `getPosition` => `position`

another naming nit, this is extracting a position from a token: `position` => `positionFromToken`

not so nit: looks like these are two functions for the price of one, because the logic is mutually exclusive between `startOfToken` true/false: `positionInToken`, `positionAtStartOfToken` 

another style nit:
```suggestion
export const position = ({ col, line, offset, text, lineBreaks, byte }: LexerToken, startOfToken = true): SourcePos => {
```

Reply:
+1 for breaking it down into two diff function - but as for the variable unpacking - is it rly nesserecy? I know the lang *can* do it but I find it results in a far to complicated function sig...